test/widgets/action_sheet_test.dart

```diff
@@ -99,6 +103,302 @@ void main() {
+      await tester.tap(unfollow);
+      await tester.pump();
+      checkUpdateUserTopicRequest(UserTopicVisibilityPolicy.none);
+    });
+
+    testWidgets('request fails with an error dialog', (tester) async {
+      await setupToTopicActionSheet(tester,
+        isChannelMuted: false,
+        visibilityPolicy: UserTopicVisibilityPolicy.followed);
+
+      connection.prepare(httpStatus: 400, json: {
+        'result': 'error', 'code': 'BAD_REQUEST', 'msg': ''});
+      await tester.tap(unfollow);
+      await tester.pumpAndSettle();
+
+      checkErrorDialog(tester, expectedTitle: 'Failed to unfollow topic');
+    });
+
+    group('check expected buttons', () {
+      final testCases = {
+        (false, UserTopicVisibilityPolicy.muted,    [unmute, follow]),
+        (false, UserTopicVisibilityPolicy.none,     [mute, follow]),
+        (false, UserTopicVisibilityPolicy.unmuted,  [mute, follow]),
+        (false, UserTopicVisibilityPolicy.followed, [mute, unfollow]),
+
+        (true,  UserTopicVisibilityPolicy.muted,    [unmuteInMutedChannel, follow]),
+        (true,  UserTopicVisibilityPolicy.none,     [unmuteInMutedChannel, follow]),
+        (true,  UserTopicVisibilityPolicy.unmuted,  [mute, follow]),
+        (true,  UserTopicVisibilityPolicy.followed, [mute, unfollow]),
+
+        (null,  UserTopicVisibilityPolicy.none,     <Finder>[]),
+      };
+
+      for (final (isChannelMuted, visibilityPolicy, buttons) in testCases) {
+        final description = 'isChannelMuted: ${isChannelMuted ?? "(not subscribed)"}, $visibilityPolicy';
+        testWidgets(description, (tester) async {
+          await setupToTopicActionSheet(tester,
+            isChannelMuted: isChannelMuted,
+            visibilityPolicy: visibilityPolicy);
+          checkButtons(buttons);
+        });
+      }
+    });
+
+    group('legacy: follow is unsupported when FL < 219', () {
+      final testCases = {
+        (false, UserTopicVisibilityPolicy.muted,    [unmute]),
+        (false, UserTopicVisibilityPolicy.none,     [mute]),
+        (false, UserTopicVisibilityPolicy.unmuted,  [mute]),
+        (false, UserTopicVisibilityPolicy.followed, [mute]),
+
+        (true,  UserTopicVisibilityPolicy.muted,    [unmuteInMutedChannel]),
+        (true,  UserTopicVisibilityPolicy.none,     [unmuteInMutedChannel]),
+        (true,  UserTopicVisibilityPolicy.unmuted,  [mute]),
+        (true,  UserTopicVisibilityPolicy.followed, [mute]),
+
+        (null,  UserTopicVisibilityPolicy.none,     <Finder>[]),
+      };
+
+      for (final (isChannelMuted, visibilityPolicy, buttons) in testCases) {
+        final description = 'isChannelMuted: ${isChannelMuted ?? "(not subscribed)"}, $visibilityPolicy';
+        testWidgets(description, (tester) async {
+          await setupToTopicActionSheet(tester,
+            isChannelMuted: isChannelMuted,
+            visibilityPolicy: visibilityPolicy,
+            featureLevel: 218);
```

Comment:
nit: use consistent standard name for a given concept
```suggestion
            zulipFeatureLevel: 218);
```